docs/arcadia-users-group/20220919-aws-s3-cli/lesson.md

```diff
@@ -0,0 +1,301 @@
+
+It should look like this:
+
+```{bash}
+aws configure
+AWS Access Key ID [None]: <YOUR_ACCESS_KEY>
+AWS Secret Access Key [None]: <YOUR_ACCESS_SECRET>
+Default region name [None]: us-west-1
+Default output format [None]: json
+```
+
+### [Advanced] When working with an EC2 instance
+
+This section is out of the scope of this workshop, but it is important to note: When working with an AWS EC2 instance you can enable S3 access by using IAM roles. Instructions for that can be found [here](https://www.notion.so/arcadiascience/Enabling-EC2-and-S3-connection-3d8b3b75441b49eaac1095eb66fbde97).
+
+## Creating test data
+
+Let's create some basic test data by copying the following commands:
+
+```{bash}
+cd ~/Desktop
+mkdir s3-test
+cd s3-test
+echo "copy" >> copy.txt
+echo "move" >> move.txt
+
+mkdir sync
+cd sync
+echo "sync 1" >> sync1.txt
+echo "sync 2" >> sync2.txt
+echo "sync 3" >> sync3.txt
+
+cd ..
+ls
+```
+
+## S3 CLI command structure
+
+The general structure of an AWS S3 CLI command follows the following structure:
+
+```{bash}
+aws s3 <COMMAND> <FLAGS> <SOURCE_PATH> <TARGET_PATH>
+```
+
+* `<COMMAND>` could be `cp`, `mv`, `sync` etc.
+* `<FLAGS>` could be something like `--dry-run` to display the operations that would be performed using the specified `<COMMAND>` without actually running them.
+* `<SOURCE_PATH>` and `<TARGET_PATH>` could be a path to a local file/directory, or an S3 file/directory path. For local files, the paths can be absolute or relative.
+
+
+As an example, something like `aws s3 cp lesson.md s3://aug-workshop-demo/lesson.md` would copy the local `lesson.md` file from my computer to the S3 bucket `aug-workshop-demo` and create a file called `lesson.md`.
+
+## S3 commands we'll work with
+
+For all the following commands, we'll be working with an S3 bucket called `aug-workshop-demo` hosted through Arcadia's AWS account. For the workshop, we'll focus on a small but useful subset of commands, but you can find the full reference documentation [here](https://docs.aws.amazon.com/cli/latest/reference/s3/).
+
+### ls
+
+In the [first shell workshop](../20220906-intro-to-shell1/lesson.md), we learned that `ls` lists contents of a directory (**l**i**s**t). It serves the same function for the S3 CLI.
+
+Let's start with listing all the Arcadia Science S3 buckets:
+
+```{bash}
+aws s3 ls
+```
+
+Now let's explore what one of these buckets looks like, starting with the `aug-workshop-demo` bucket.
+
+```{bash}
+aws s3 ls s3://aug-workshop-demo
+```
+
+If everything went correctly, you should see the `demo` folder under the `aug-workshop-demo` bucket.
+
+### cp
+
+In the [second shell workshop](../20220912-intro-to-shell2/lesson.md), we learned that `cp` copies a file or directory. The S3 command also serves a similar purpose. Let's try it out by trying to copy the file we created earlied `copy.txt` to the S3 bucket under a folder with your initials. For me, this would be in `s3://aug-workshop-demo/fmc/`.
```

Comment:
```suggestion
In the [second shell workshop](../20220912-intro-to-shell2/lesson.md), we learned that `cp` copies a file or directory. The S3 command also serves a similar purpose. Let's try it out by trying to copy the file we created earlier `copy.txt` to the S3 bucket under a folder with your initials. For me, this would be in `s3://aug-workshop-demo/fmc/`.
```